### ISSUE TITLE
some question about batchnorm

### ISSUE THREAD

**Issue by yangjun-z** [ceciliavision/perceptual-reflection-removal#6]
I notice you use batch norm both  in discriminator and the generator,as shown in the code

discriminator.py:
```
def batchnorm(input):
    with tf.variable_scope("batchnorm"):
        # this block looks like it has 3 inputs on the graph unless we do this
        input = tf.identity(input)

        channels = input.get_shape()[3]
        offset = tf.get_variable("offset", [channels], dtype=tf.float32, initializer=tf.zeros_initializer())
        scale = tf.get_variable("scale", [channels], dtype=tf.float32, initializer=tf.random_normal_initializer(1.0, 0.02))
        mean, variance = tf.nn.moments(input, axes=[0, 1, 2], keep_dims=False)
        variance_epsilon = 1e-5
        normalized = tf.nn.batch_normalization(input, mean, variance, offset, scale, variance_epsilon=variance_epsilon)
        return normalized
```

main.py:

```
def nm(x):
    w0=tf.Variable(1.0,name='w0')
    w1=tf.Variable(0.0,name='w1')
    return w0*x+w1*slim.batch_norm(x)

net=slim.conv2d(input,channel,[1,1],rate=1,activation_fn=lrelu,normalizer_fn=nm,weights_initializer=identity_initializer(),scope='g_conv0')
...
```

but it seems you do nothing with moving_mean and moving_variance,it makes me alittle confused.

**Comment by Vandermode** [comment]
Actually, there are two implementation of batch normalization. The first is the standard batch normalization using `tf.nn.batch_normalization`.  And the second is the adaptive batch normalization which is described in "Fast image processing via fully-convolutional neural network, ICCV 17".

**Comment by yangjun-z** [comment]
Oh i see,sorry for my careless.But what about the batchnorm in discriminator? it is tf.nn.batch_normalization, @Vandermode ,Actually,i am not very familiar with tensorflow,i use pytorch most of the time,so if i ask a basic question,please forgive me.
